dGame/dBehaviors/BehaviorContext.cpp

```diff
@@ -183,6 +186,21 @@ void BehaviorContext::SyncCalculation(const uint32_t syncId, const float time, B
 	this->syncEntries.push_back(entry);
 }
 
+void BehaviorContext::UpdatePlayerSyncs(float deltaTime) {
```

Comment:
This does account for two entries timing out on the same frame, right? I could see it, just an odd way to iterate.

Reply:
Yes, which is why it iterates so oddly.  If you use range based you are modifying the vector while in a range based vector which invalidates all iterators past the one you erased, hence decrement time, check if we should remove it and if we should, just remove it otherwise go to the next element